oeps/processes/oep-0053-proc-package-release.rst

```diff
@@ -0,0 +1,153 @@
++---------------+--------------------------------------------------------------+
+| Last-Modified | 2021-04-07                                                   |
++---------------+--------------------------------------------------------------+
+| Authors       | Jawayria Hashmi <jhashmi@edx.org>,                           |
+|               | Aarif <aarif@edx.org>                                        |
++---------------+--------------------------------------------------------------+
+| Arbiter       |                                                              |
++---------------+--------------------------------------------------------------+
+| Status        | Draft                                                        |
++---------------+--------------------------------------------------------------+
+| Type          | Process                                                      |
++---------------+--------------------------------------------------------------+
+| Created       | 2021-04-07                                                   |
++---------------+--------------------------------------------------------------+
+| Review Period |                                                              |
++---------------+--------------------------------------------------------------+
+| References    |                                                              |
++---------------+--------------------------------------------------------------+
+
+Abstract
+========
+
+This document proposes a standard package release process to be followed across the Open edX community.
+
+Motivation
+==========
+
+We recently shifted the CI from Travis to Github Actions which led us to move the package release process to Github Actions as well. During this process,
+we observed some problems in the current package release process. The package release process is comprised of several manual steps.
+Once the changes are pushed, we've to update the changelogs, bump the version number then create a tag or Github release which finally triggers the final package release.
+As all these steps need to be done manually, sometimes things get missed. For instance, you made changes, bumped the version, and created a release but forgot
+to add changelogs. Moreover, we prefer to publish releases on tags but also want to keep Github releases aligned with PyPI releases so that release history
+could be found in the GitHub repository as well.
+
+To solve these above-mentioned issues, we have automated the package release process. Initially, we looked for packages that could do the job but the
+available packages don't provide much control so we created a pipeline using a set of Github workflows to achieve the goal. By using the release process
+proposed in the OEP we’ll be able to achieve the following:
+
+* Just push the changes and the release pipeline will automatically take care of the rest of the process.
```

Comment:
This bulleted list is very specific to python packages.

The npm workflow is actually quite different, in that on push to master, `semantic-release` examines the merged commits for conventional commits and generates a semantic version number, an npm release, a Github release, and a tag based on what it finds.  The tag is output, not the input, and the whole process is automated on pushing commits to master.

I wonder if this OEP should be specific to python packages?  Either that or we have sections specific to the two different languages and workflows.

Reply:
Hey @davidjoy thanks for taking time out to review the OEP. I had a discussion about this with Jeremy and we came to the conclusion that this OEP will be specific to python packages and we'll write a separate OEP or confluence document for JS packages.